src/store/services.ts

```diff
@@ -3,22 +3,68 @@ import { TAuthor, TCourse, TUser, TServerReturn } from 'src/shared.types';
 async function doPost(params: {
 	url: string;
 	payload: object;
+	token?: string;
 }): Promise<TServerReturn> {
-	const { url, payload } = params;
+	const { url, payload, token } = params;
 	const response = await fetch(url, {
 		method: 'POST',
 		body: JSON.stringify(payload),
 		headers: {
+			Authorization: token,
```

Comment:
[optional] it's better not to pass token explicitly all the time. Just retrieve it here from localStorage. It will greatly simplify you code related to api call.